packages/wallet/backend/src/socket/service.ts

```diff
@@ -0,0 +1,62 @@
+import { env, type Env } from '@/config/env'
+import { Logger } from 'winston'
+import socketIo from 'socket.io'
+import http from 'http'
+import { unsealData } from 'iron-session'
+import { AccountService } from '@/account/service'
+
+interface ISocketService {
+  init(args: http.Server): void
+  emitAccountsUpdateById(id: string): void
+}
+
+interface SocketServiceDependencies {
+  logger: Logger
+  env: Env
+  accountService: AccountService
+}
+
+export class SocketService implements ISocketService {
+  private io: socketIo.Server | null = null
+
+  constructor(private deps: SocketServiceDependencies) {}
+
+  init(server: http.Server) {
+    this.io = new socketIo.Server(server, {
+      cors: {
+        origin: [
+          'http://localhost:4003',
+          `https://${this.deps.env.RAFIKI_MONEY_FRONTEND_HOST}`
+        ],
+        credentials: true
+      }
+    })
+    this.deps.logger.info(`Socket Server is started...`)
+
+    this.io.on('connection', async (socket) => {
+      const cookie = socket.request.headers['cookie']
```

Comment:
as the token is sent in cookies it should be possible to use the express middleware
https://socket.io/docs/v4/middlewares/#:~:text=Compatibility%20with%20Express%20middleware%E2%80%8B,really%20compatible%20with%20Express%20middlewares.